client/cli/src/lib.rs

```diff
@@ -215,487 +133,26 @@ where
 	Ok(T::from_clap(&matches))
 }
 
-/// A helper function that initializes and runs the node
-pub fn run<F, G, E, FNL, FNF, SL, SF>(
-	mut config: Configuration<G, E>,
-	run_cmd: RunCmd,
-	new_light: FNL,
-	new_full: FNF,
-	spec_factory: F,
-	version: &VersionInfo,
-) -> error::Result<()>
-where
-	F: FnOnce(&str) -> Result<Option<ChainSpec<G, E>>, String>,
-	FNL: FnOnce(Configuration<G, E>) -> Result<SL, sc_service::error::Error>,
-	FNF: FnOnce(Configuration<G, E>) -> Result<SF, sc_service::error::Error>,
-	G: RuntimeGenesis,
-	E: ChainSpecExtension,
-	SL: AbstractService + Unpin,
-	SF: AbstractService + Unpin,
-{
-	init(&run_cmd.shared_params, version)?;
-	init_config(&mut config, &run_cmd.shared_params, version, spec_factory)?;
-	run_cmd.run(config, new_light, new_full, version)
-}
-
-/// A helper function that initializes and runs any of the subcommand variants of `CoreParams`.
-pub fn run_subcommand<F, G, E, B, BC, BB>(
-	mut config: Configuration<G, E>,
-	subcommand: Subcommand,
-	spec_factory: F,
-	builder: B,
-	version: &VersionInfo,
-) -> error::Result<()>
-where
-	F: FnOnce(&str) -> Result<Option<ChainSpec<G, E>>, String>,
-	B: FnOnce(Configuration<G, E>) -> Result<BC, sc_service::error::Error>,
-	G: RuntimeGenesis,
-	E: ChainSpecExtension,
-	BC: ServiceBuilderCommand<Block = BB> + Unpin,
-	BB: sp_runtime::traits::Block + Debug,
-	<<<BB as BlockT>::Header as HeaderT>::Number as std::str::FromStr>::Err: std::fmt::Debug,
-	<BB as BlockT>::Hash: std::str::FromStr,
-{
-	let shared_params = subcommand.get_shared_params();
-	init(shared_params, version)?;
-	init_config(&mut config, shared_params, version, spec_factory)?;
-	subcommand.run(config, builder)
-}
```

Comment:
@bkchr  since you're here already I'm showing you this.

As you can see I'm dropping all of this. I know you said it would be best to have to functions directly on the commands and that's what I did. Every command now has a `update_config` and a `run` functions. They are separated to allow the developer to force/override things after parsing the command line. Check in bin/node/cli/src/commands.rs